wordpress-mobile/WordPress-Android
# Issue/8670 site creation errors segments

Fixes #8670 


1. Removes slide-in animation for the first screen of the Site Creation flow

2. UiState refactoring - the structure we came up with felt like an overkill and just complicated things. I also changed my mind regarding having ideally no conditional statements in view layer. As long as they are straightforward I think it's ok to have them. So if you don't mind I decided to go with simpler approach -> I've separated the UiState into two sub states -> Content/Error -> if a value in the UiState is null ->  we hide the view. I'm sorry for keep doing these changes, but I realized how complicated and not-flexible it is while I was introducing the error states.

3. Introduces TEST_DISPATCHER which doesn't wait on `delay(n)` calls.

4. Adds support for "No Connection" error

To test:
- set the `NEW_SITE_CREATION_ENABLED` build time feature flag to `true`
1. Start the SiteCreation flow
2. Wait until segments loads
3. Select a segment
4. Make sure you are navigated to the Verticals screen
-------
1. Turn on Airplane mode
2. Start the SiteCreation flow
3. Notice the "No connection" error message
4. Click on Retry
5. Notice the "No connection" error message
6. Turn off airplane mode
7. Click on Retry
8. Make sure the segments are shown
--------
- add `event.error = FetchSegmentsError(GENERIC_ERROR)` to `FetchSegmentsUseCase.onSiteCategoriesFetched(...)`
1. Start the SiteCreation flow
2. Notice the "There was a problem" error
--------
![untitled](https://user-images.githubusercontent.com/2261188/49507832-950e1d80-f881-11e8-9354-0dc9baf17110.png)
